src/js/hooks/cycleCount/useCycleCountFilters.js

```diff
@@ -71,49 +73,54 @@ const useCycleCountFilters = () => {
   };
 
   const initializeDefaultFilterValues = async () => {
-    const queryProps = queryString.parse(history.location.search);
-
-    const defaultValues = Object.keys(cycleCountFilterFields)
-      .reduce((acc, key) => ({ ...acc, [key]: '' }), { tab: queryProps.tab });
-
-    if (queryProps.negativeQuantity) {
-      defaultValues.negativeQuantity = queryProps.negativeQuantity;
+    setIsLoading(true);
+
+    try {
+      const queryProps = queryString.parse(history.location.search);
+      const defaultValues = Object.keys(cycleCountFilterFields)
+        .reduce((acc, key) => ({ ...acc, [key]: '' }), { tab: queryProps.tab });
+
+      if (queryProps.negativeQuantity) {
+        defaultValues.negativeQuantity = queryProps.negativeQuantity;
+      }
+
+      if (queryProps.dateLastCount) {
+        defaultValues.dateLastCount = queryProps.dateLastCount;
+      }
+
+      const [
+        categoryList,
+        tagList,
+        catalogList,
+        classificationList,
+        binList,
+      ] = await Promise.all([
+        fetchProductsCategories(),
+        fetchProductsTags({ hideNumbers: true }),
+        fetchProductsCatalogs({ hideNumbers: true }),
+        fetchProductClassifications(currentLocation?.id),
+        fetchBins(currentLocation?.id),
+      ]);
+
+      setSelectOptions({
+        categories: categoryList,
+        catalogs: catalogList,
+        tags: tagList,
+        internalLocations: binList,
+        abcClasses: classificationList,
+      });
+
+      defaultValues.catalogs = setDefaultValue(queryProps.catalogs, catalogList);
+      defaultValues.tags = setDefaultValue(queryProps.tags, tagList);
+      defaultValues.categories = setDefaultValue(queryProps.categories, categoryList);
+      defaultValues.internalLocations = setDefaultValue(queryProps.internalLocations, binList);
+      defaultValues.abcClasses = setDefaultValue(queryProps.abcClasses, classificationList);
+
+      setDefaultFilterValues(defaultValues);
+      setFiltersInitialized(true);
```

Comment:
can't you use filtersInitialized instead of isLoading? Looks like it handles similar cases,